### ISSUE TITLE
ValueError: The truth value of a DataFrame is ambiguous.

### ISSUE THREAD

**Issue by dserban** [J535D165/recordlinkage#22]
Code:
```
from pandas import DataFrame as DF
import jellyfish
import recordlinkage

names = \
    [ {'name': 'Daniel'}
    , {'name': 'Danieel'}
    , {'name': 'Daaniel'}
    , {'name': 'Daniiel'}
    , {'name': 'Alex'}
    , {'name': 'George'}
    , {'name': 'Laura'}
    , {'name': 'Mary'}
    ]
names_df = DF(names)
names_df.index.name = 'aaa'
ps = recordlinkage.Pairs(names_df)
pairs = ps.full()
compare_cl = recordlinkage.Compare(pairs, names_df, names_df)
print compare_cl.fuzzy('name', 'name', method='jarowinkler', threshold=0.85)
```
The error I'm getting:
```
Traceback (most recent call last):
  File "rl.py", line 20, in <module>
    print compare_cl.fuzzy('name', 'name', method='jarowinkler', threshold=0.85)
  File "/opt/anaconda/lib/python2.7/site-packages/pandas/core/base.py", line 47, in __str__
    return self.__bytes__()
  File "/opt/anaconda/lib/python2.7/site-packages/pandas/core/base.py", line 59, in __bytes__
    return self.__unicode__().encode(encoding, 'replace')
  File "/opt/anaconda/lib/python2.7/site-packages/pandas/core/series.py", line 984, in __unicode__
    max_rows=max_rows)
  File "/opt/anaconda/lib/python2.7/site-packages/pandas/core/series.py", line 1025, in to_string
    dtype=dtype, name=name, max_rows=max_rows)
  File "/opt/anaconda/lib/python2.7/site-packages/pandas/core/series.py", line 1053, in _get_repr
    result = formatter.to_string()
  File "/opt/anaconda/lib/python2.7/site-packages/pandas/formats/format.py", line 224, in to_string
    fmt_index, have_header = self._get_formatted_index()
  File "/opt/anaconda/lib/python2.7/site-packages/pandas/formats/format.py", line 206, in _get_formatted_index
    have_header = any(name for name in index.names)
  File "/opt/anaconda/lib/python2.7/site-packages/pandas/core/generic.py", line 892, in __nonzero__
    .format(self.__class__.__name__))
ValueError: The truth value of a DataFrame is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
```
Am I doing something wrong?

**Comment by J535D165** [comment]
Hello dserban,

Thanks for your feedback and sorry for this bug. I found the solution and will push an update tomorrow.

**Comment by J535D165** [comment]
The bug is solved in version 0.4.0 (install with ``pip install recordlinkage``)

Two things about your example: 
- When you are using Python 2.x, then the strings need to be unicode. 
- ```compare.fuzzy``` is now ```compare.string```. (both work btw)

``` python

from pandas import DataFrame as DF
import jellyfish
import recordlinkage

names = \
    [ {'name': u'Daniel'}
    , {'name': u'Danieel'}
    , {'name': u'Daaniel'}
    , {'name': u'Daniiel'}
    , {'name': u'Alex'}
    , {'name': u'George'}
    , {'name': u'Laura'}
    , {'name': u'Mary'}
    ]
names_df = DF(names)
names_df.index.name = 'aaa'
ps = recordlinkage.Pairs(names_df)
pairs = ps.full()
compare_cl = recordlinkage.Compare(pairs, names_df, names_df)
print compare_cl.string('name', 'name', method='jarowinkler', threshold=0.85)

```

**Comment by Shonexu** [comment]
Hi All, 

i encountered the same problem using Python 3.7 and recordlinkage 0.14 . Your advice would be highly appreciated

below is the detail of the error message
=============================
ValueError                                Traceback (most recent call last)
<ipython-input-16-181ee39d989a> in <module>
----> 1 compare = rl.Compare(candidate_links, bib, grid)

~\AppData\Local\Continuum\anaconda3\lib\site-packages\recordlinkage\base.py in __init__(self, features, n_jobs, indexing_type, **kwargs)
    540 
    541         # public
--> 542         if n_jobs == -1:
    543             self.n_jobs = cpu_count()
    544         else:

~\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\generic.py in __nonzero__(self)
   1476         raise ValueError("The truth value of a {0} is ambiguous. "
   1477                          "Use a.empty, a.bool(), a.item(), a.any() or a.all()."
-> 1478                          .format(self.__class__.__name__))
   1479 
   1480     __bool__ = __nonzero__

ValueError: The truth value of a DataFrame is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
